FogCreek/Glitch-Community
# Efficacious deposit: start work on getting "my stuff" button inside popovers

UPDATE: this is currently a WIP, as the api was recently deployed and I merged in some assets for it to use... check this thing out later I guess

## Links
* https://efficacious-deposit.glitch.me
* https://app.zenhub.com/workspaces/community-5d07d1359dee6059a9688176/issues/fogcreek/glitch-community-work/170
* designs: https://www.notion.so/glitch/My-Stuff-Default-Collection-f9eb3c8196c6431596fba526fc5b91c2#26f548132700419bbbbb9cb30e284c1f

## GIF/Screenshots:
![Screen Shot 2019-07-24 at 5 33 33 PM](https://user-images.githubusercontent.com/6620164/61830496-4970aa00-ae39-11e9-9a6f-388edd9eaeed.png)
![Screen Shot 2019-07-24 at 5 34 08 PM](https://user-images.githubusercontent.com/6620164/61830497-4970aa00-ae39-11e9-8dd3-d6dfa3b80c4a.png)

## Changes:
- adds a My Stuff collection to the AddProject to Collection Popovers. 
    - if it exists in the database we show that one
    - if it doesn't exist in the database we mock it out and create one once you click on it.
- adds a My Stuff collection to our User Pages
    - if it exists in the database we show that one
    - if it doesn't exist yet we mock it out and create one once you click on it. 
- removes curators from add to collection popovers in the user collection tab, (we still show them for the team collections tab)

## How To Test:
QA List (tests coming in another pr...maybe this one? idk haven't decided yet.)
When My Stuff is not enabled (THIS IS MOST IMPORTANT since we're not releasing yet. We'll have time for more qa later for the features, but I want to ensure nothing I'm deploying now breaks what's in prod)
- [x] I can add a project to collection fine using the popover
- [x] I don't see any references to My Stuff on my user page 
- [x] I don't see any references to My Stuff on the Add to Collection Popover 

When My Stuff is enabled (efficacious-deposit.glitch.me/secret)
You'll likely want to create multiple test accounts for this, I recommend signing in with email using this format name+test1@glitch.com 
- [x]  With a brand new user: On my user page I see a My Stuff collection even though I've never made a collection before. If I click on it I'm brought to a my stuff collection
- [x] With a brand new user: On the add project to collection popover I see a "My Stuff" collection even though I've never made a collection before. If I click on it creates one for me and adds the project to the collection. 
- [x] With a brand new user: I create a My Stuff collection as described above but ensure it has no projects inside it. As a logged out or different user if I go to that previous user's page and do not see a `My Stuff` collection. If I log back in and add a project to it, the logged out user can now see the collection
- [x] When I search for a collection in the add project to popover collection, My Stuff doesn't show up when there's a non matching search query, (When I search for a collection and I haven't created My Stuff yet it won't appear in search results)
- [x] when you click the teams tab in the add project to collection popover it shouldn't show up. 

## Feedback I'm looking for:
- bugsss.
